### PR TITLE
[DT-788] Footer: Apply for Access to selected Studies and Datasets

### PR DESCRIPTION
**Pull request by rjohanek**
### Addresses
https://broadworkbench.atlassian.net/browse/DT-788

### Summary
Moves the apply for access button to the new footer component that is optionally rendered if any datasets/studies are selected. Also displays text about the number of datasets and studies selected.

https://github.com/user-attachments/assets/d53f984f-38e8-4b0d-8ed8-4e2fa177c916

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by lstrano-broad** [comment]
LGTM - thx

**Review by cinyecai — APPROVED** [review]
Lgtm, thanks for including the video!

File: src/components/data_search/DatasetSearchFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    zIndex: 999,
+    width: '100vw',
+    height: 60,
+    display: 'flex',
+    justifyContent: 'flex-end',
+    backgroundColor: 'white',
+    border: '1px solid #DEDEDE',
+    alignItems: 'center'
+  }}>
+    {selectedDatasets.length > 1 && selectedStudies.length > 1 && (<div style={{paddingRight: 15}}> {selectedDatasets.length} Datasets selected from {selectedStudies.length} Studies </div>)}
```
Comment (fboulnois) [inline src/components/data_search/DatasetSearchFooter.tsx]: How about something like:

```
// at top of component
const datasetText = selectedDatasets.length > 1 ? 'datasets' : 'dataset';
const studyText = selectedStudies.length > 1 ? 'studies' : 'study';

{selectedDatasets.length > 0 && (<div>{selectedDatasets.length} {datasetText} selected from {selectedStudies.length} {studyText}</div>)}

**Review by fboulnois — COMMENTED** [review]
One early suggestion:

File: src/components/dac_dataset_table/DACDatasetConstants.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import {Styles} from '../../libs/theme';
+
+export const styles = {
```
Comment (rushtong) [inline src/components/dac_dataset_table/DACDatasetConstants.tsx:3]: Another approach to this is to use modular css. We have several examples in the codebase for this. If the style objects are being manipulated by components, then I think that approach wouldn't work.

**Review by rushtong — APPROVED** [review]
Looks nice 👍🏽

File: src/components/dac_dataset_table/DACDatasetTableCellData.jsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import style from '../../pages/DACDatasets.module.css';
-import {styles} from './DACDatasetsTable';
+import {styles} from './DACDatasetConstants';
```
Comment (rjohanek) [inline src/components/dac_dataset_table/DACDatasetTableCellData.jsx:3]: removing circular dependency discovered when testing

File: src/components/data_search/DatasetSearchTableConstants.tsx [hunk 4]
```diff
@@ -79,7 +79,7 @@ export const makeStudyTableHeaders = (datasets: DatasetTerm[], selected: number[
   const selectableDatasetIds = datasets.filter(isSelectable).map(dataset => dataset.datasetId);
   return [
     {
-      label: <Checkbox checked={datasets.length === selected.length}
+      label: <Checkbox id='header-checkbox' checked={datasets.length === selected.length}
```
Comment (rjohanek) [inline src/components/data_search/DatasetSearchTableConstants.tsx:82]: added the id here so the element is easily retrievable in testing

**Review by rushtong — APPROVED** [review]
Looks very nice, thank you 👍🏽

**Review by fboulnois — APPROVED** [review]
looks great 👍

**Pull request merged**